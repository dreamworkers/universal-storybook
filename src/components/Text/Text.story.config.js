/**
 * Common story configuration
 */

export const ArgTypesText = {
  content: {
    control: {
      type: "text",
    },
  },
  color: { control: "color" },
};

export const ArgsText = {
  color: null,
};

export const ArgsTextPrimary = {
  content: `Globular star cluster Apollonius of Perga courage of our questions vastness is bearable only through love vanquish the impossible cosmos? Are creatures of the cosmos hydrogen atoms Rig Veda dream of the mind's eye astonishment a still more glorious dawn awaits? From which we spring a mote of dust suspended in a sunbeam star stuff harvesting star light kindling the energy hidden in matter a mote of dust suspended in a sunbeam network of wormholes?`,
  ...ArgsText,
};

/**
 * Web core story configuration
 */

export const ArgTypesTextHtml = {
  ...ArgTypesText,
  render: {
    control: false,
  },
};

export const ArgsTextHtml = {
  render: "html",
};

export const ArgsTextPrimaryHtml = {
  ...ArgsTextPrimary,
  ...ArgsTextHtml,
};
