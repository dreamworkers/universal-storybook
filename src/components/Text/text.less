@import "../../globals/tokens.less";

p {
  font-family: @font-family-base;
  font-weight: 400;
  line-height: 1.4;
}
