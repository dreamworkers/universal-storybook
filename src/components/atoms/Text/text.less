@import '@Styles/inc/tokens.less';

p {
  font-family: @family-font-base;
  font-weight: @weight-font-normal;
  line-height: @line-height-font-base;
}
