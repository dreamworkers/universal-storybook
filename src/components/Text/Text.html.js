import "./text.less";

const getClassName = ({ size }) => {
  const classes = ["text"];
  if (size) classes.push(`text--${size}`);
  return classes.join(" ");
};

const getColor = (color) => (color ? ` style="color: ${color}"` : "");

export const TextStory = ({ content, color, size }) => {
  return `<p className="${getClassName({
    size,
  })}"${getColor(color)}>${content}</p>`;
};
