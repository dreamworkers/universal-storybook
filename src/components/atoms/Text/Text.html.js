import "./text.less";

const getColor = (color) => (color ? ` style="color: ${color}"` : "");

export const TextHtml = ({ content, color }) => {
  return `<p${getColor(color)}>${content}</p>`;
};
