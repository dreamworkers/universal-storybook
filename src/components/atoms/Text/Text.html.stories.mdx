import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';

import { TextHtml } from './Text.html';
import { ArgTypesTextHtml, ArgsTextPrimaryHtml } from './Text.story.config';

<!-- Stories -->

<Meta
  title="HTML/Atoms/Text"
  parameters={{
    jest: ['Text.html.test.js'],
  }}
/>

<Canvas withSource="open">
  <Story
    name="Basic text"
    argTypes={ArgTypesTextHtml}
    args={ArgsTextPrimaryHtml}
  >
    {TextHtml.bind({})}
  </Story>
</Canvas>

<!-- Documentation -->

# Text

## Basic Text

<p>Text</p>

```
<p>Text</p>
```
