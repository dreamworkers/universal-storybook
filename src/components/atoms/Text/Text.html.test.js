import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { TextHtml } from './Text.html';

configure({ adapter: new Adapter() });

describe('Text (HTML component)', () => {
  describe('will render', () => {
    it('creates a text component', () => {
      const props = {
        content: 'text content',
      };
      const tree = shallow(<TextHtml {...props} />);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Will take in multiple Props', () => {
    let tree;

    beforeAll(() => {
      const props = {
        color: null,
        content: 'text content',
      };
      tree = shallow(<TextHtml {...props} />);
    });

    it('Will have a child text', () => {
      expect(tree.text()).toEqual('<p>text content</p>');
    });
  });
});
