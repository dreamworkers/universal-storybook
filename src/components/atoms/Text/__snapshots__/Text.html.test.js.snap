// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text (HTML component) will render creates a text component 1`] = `ShallowWrapper {}`;
